### ISSUE TITLE
Can´t get Bot34.1 to work

### ISSUE THREAD

**Issue by Larrush** [Betalord/BHBot#46]
```
<2019/03/27 13:35:43> BHBot v34.1 started.
<2019/03/27 13:35:43> Found 18 familiar cue(s) with old format.
<2019/03/27 13:35:43> Found 53 familiar cue(s) with new format.
<2019/03/27 13:35:43> Idle mode started, no actions will be taken
<2019/03/27 13:35:43> Found Chromium in C:\Users\***\AppData\Local\Chromium\Application\chrome.exe
<2019/03/27 13:35:43> Found chromedriver in C:\Users\***\Favorites\Downloads\BHBot-master\chromedriver.exe
<2019/03/27 13:35:43> Found cues in C:\Users\***\Favorites\Downloads\BHBot-master\cues
<2019/03/27 13:35:43> Driver started succesfully
Starting ChromeDriver 2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90) on port 38171
Only local connections are allowed.
<2019/03/27 13:35:49> Something went wrong with driver restart. Will retry in a few minutes... (sleeping)
org.openqa.selenium.WebDriverException: unknown error: cannot create temp dir for unpacking extensions
  (Driver info: chromedriver=2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90),platform=Windows NT 10.0.17134 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 61 milliseconds
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: '****', ip: '192.168.0.***', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_201'
Driver info: driver.version: ChromeDriver
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:214)
        at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:166)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$new$0(JsonWireProtocolResponse.java:53)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$getResponseFunction$2(JsonWireProtocolResponse.java:91)
        at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:122)
        at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.util.Spliterators$ArraySpliterator.tryAdvance(Unknown Source)
        at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:125)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:73)
        at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:136)
        at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:548)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:212)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:130)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:181)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:168)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:157)
        at MainThread.connectDriver(MainThread.java:752)
        at MainThread.restart(MainThread.java:881)
        at MainThread.run(MainThread.java:1038)
        at java.lang.Thread.run(Unknown Source)
```

I always get this error when opening the new version of BHBot.
Folder looks like this, in case its an obvious mistake I am missing.
![Untitled](https://user-images.githubusercontent.com/47528224/55076380-89936c00-5095-11e9-9124-a9110a45164c.jpg)

Version 34 works perfectly fine, it´s just when I upgrade to the "auto chromedriver version" that I get this.

**Comment by ilpersi** [comment]
Can you please try to move BHBot out of the Dowload folder? Try something like "c:\BHBot" and let me know if the issue persists.

**Comment by Larrush** [comment]
Yep it persists.
The first time I had it on my desktop and the second time I left it in the download folder.
But I can try it again tomorrow and tell you if I made it work :)

**Comment by Larrush** [comment]
```
<2019/03/29 12:27:12> BHBot v34.1 started.
<2019/03/29 12:27:12> Found 20 familiar cue(s) with old format.
<2019/03/29 12:27:12> Found 44 familiar cue(s) with new format.
<2019/03/29 12:27:12> Deprecated setting detected: autoBribe. Ignoring it.
<2019/03/29 12:27:12> Settings loaded from file
<2019/03/29 12:27:12> Character: ***
<2019/03/29 12:27:12> Found Chromium in C:\Users\**\AppData\Local\Chromium\Application\chrome.exe
<2019/03/29 12:27:12> Found chromedriver in C:\Users\**\Desktop\BHBot-34.1\chromedriver.exe
<2019/03/29 12:27:12> Found cues in C:\Users\**\Desktop\BHBot-34.1\cues
<2019/03/29 12:27:12> Driver started succesfully
Starting ChromeDriver 2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90) on port 20956
Only local connections are allowed.
<2019/03/29 12:27:13> Something went wrong with driver restart. Will retry in a few minutes... (sleeping)
org.openqa.selenium.WebDriverException: unknown error: cannot create temp dir for unpacking extensions
  (Driver info: chromedriver=2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90),platform=Windows NT 10.0.17134 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 52 milliseconds
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: '**PC', ip: '192.168.0.***', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_201'
Driver info: driver.version: ChromeDriver
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:214)
        at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:166)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$new$0(JsonWireProtocolResponse.java:53)
        at org.openqa.selenium.remote.JsonWireProtocolResponse.lambda$getResponseFunction$2(JsonWireProtocolResponse.java:91)
        at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:122)
        at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
        at java.util.Spliterators$ArraySpliterator.tryAdvance(Unknown Source)
        at java.util.stream.ReferencePipeline.forEachWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(Unknown Source)
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.util.stream.FindOps$FindOp.evaluateSequential(Unknown Source)
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.util.stream.ReferencePipeline.findFirst(Unknown Source)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:125)
        at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:73)
        at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:136)
        at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
        at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:548)
        at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:212)
        at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:130)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:181)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:168)
        at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:157)
        at MainThread.connectDriver(MainThread.java:752)
        at MainThread.restart(MainThread.java:881)
        at MainThread.run(MainThread.java:1038)
        at java.lang.Thread.run(Unknown Source)
```

I tried it again, and put it on my desktop, but I still got the same error message.

I thought about why it isn´t working and my PC seems to have a weird issue when programs try to use run as administrator.
Sometimes when I want to install a certain program on my pc, he asks for permission as usual and if I say yes it gives me this error code 
![Untitled](https://user-images.githubusercontent.com/47528224/55230082-0867e080-521f-11e9-8bf8-24cafbdea6f3.jpg)

Now this usually isn´t a problem, since If I run as admin manually I can just overlook this issue.

I looked on google for a solution for this, but couldn´t really find a working solution.
But if this might be the reason due to BHBot not beeing able to start chromedriver as admin, I´ll look again into it.

**Comment by Larrush** [comment]
Ok, yeah I am an idiot. 
This is the issue. If I open chromedriver without manually starting it as admin, I get the same error message.

Ooh darn, in that case I´ll try to look into it this weekend, wish me luck xD 

Can you just do me a favor? 
I didn´t realise the first time I posted the code section how much valuable information is in that code, so is there a possibility to delete this Thread or at least my first comments+edit? 

I didn´t find anything where I could delete messages

**Comment by ilpersi** [comment]
I am not sure I understand what your problem is, BTW with v34.1 you are not supposed to start chromedriver.exe in any way (neither as admin, nor as user). You just start the BHBot.jar and the bot will run chromedriver by itself.

I am not able to delete this post, what you can do is delete the revisions of your first post and just leave the last version online. I did it for you and you are also able to do it without my intervention.

**Comment by bbicudo** [comment]
@Larrush , about the error with running as admin, do u receive this error message every time you run something as admin?

Try running mmc.exe from a non administrative command prompt and see if you receive any error message.

**Comment by Larrush** [comment]
@ilpersi nono I meant that when I start BHBot 34.1 the bot tries to start chromedriver as admin which he can't since I have this random issue on my pc with admin rights in general.

(If I start BHBot 34 I have to start chromedriver with the "start as admin" function.
If I only open it normal and get asked if I want to start as admin, I get the same message as with BHBot 34.1)

This seems to be related to temp folder accesability on windows 10 as far as I understand it.

And since I can't start a .bat as admin I cant work around this issue as I usually would.

Therefore I need to find a solution to fix my pc's Admin rights and not BHBot.
This Issue is windows' fault and not the bots!

@bbicudo I'll try that tomorrow, cause I am in work until midnight but thanks for the tip!

Yes it does about 90% of the time. Just sometimes he accepts the usual admin command.
Most of the times I have to do right-click- start as admin and then it works.

**Comment by Larrush** [comment]
@bbicudo I opened it now, and so far it didn´t give me any message.
It opened the management console. 

As far as I found out about my issue is that there is a problem with rights for temporary files, which was and is a problem in Win10 and Win7 for some.
It´s supposed to be fixed by opening the Security tab from the temp folder and change restrictions, which I can´t do for whatever reason.
And at that point, I didn´t care at the time I was trying to fix it. 
But since I need to fix it now, I´ll look into that tomorrow! 
If you have any suggestions for me, I´ll try it for sure ;) Or if I can do anything about it in the mmc.exe ^^ 

Also thanks to @ilpersi for deleting my first posts. I didn´t see that.
Btw. sorry for posting and discussing a non bot related issue here! 
Since this has nothing to do with the way the bot works :)

**Comment by ilpersi** [comment]
Let me know if you are able to fix, we may consider a new parameter in the ini file to make the chromedriver autostart optional

**Comment by bbicudo** [comment]
@Larrush i asked about mmc.exe because some Windows users are reporting problems with components asking for admin privileges even when they're admins.

It's a known problem with crypto services which is affecting both W7 and W10. But as you opened mmc.exe without any issues, that's not your problem. 

About opening the security tab to change restrictions on TEMP folder, try 

`takeown /f <path_to_folder\foldername> /r /d y`

on an elevated command prompt. After you take ownership on TEMP, change whatever you need. After that, go to Security > Advanced and give ownership back to System if you want to (not really necessary).

Hope that helps :)

**Comment by ilpersi** [comment]
I've released [v34.2alpha2](https://github.com/Betalord/BHBot/releases/tag/v34.2a2) with a new ini option to make the chromedriver auto start optional.

Add this line to your settings.ini to revert to the v34 behavior:

`autoStartChromeDriver 0`

**Comment by Larrush** [comment]
Thanks so much again! 
So with that update I think this Issue can be seen as closed :D 

That´s nice to hear, but also do I appreciate the amount of work you have to put in, to make this such a good bot!
